plugins/importexport/crossref/filter/ArticleCrossrefXmlFilter.inc.php

```diff
@@ -0,0 +1,278 @@
+		// append galleys files and collection nodes to the DOI data node
+		// galley can contain several files
+		$articleGalleyDao = DAORegistry::getDAO('ArticleGalleyDAO');
+		$galleys = $articleGalleyDao->getBySubmissionId($submission->getId());
+		import('lib.pkp.classes.submission.SubmissionFile'); // SUBMISSION_FILE_... constants
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
+		$submissionGalleyFiles = array();
+		// get immediatelly also supplementary files for component list
+		$componentFiles = array();
+		while ($galley = $galleys->next()) {
+			$galleyFiles = $submissionFileDao->getLatestRevisionsByAssocId(ASSOC_TYPE_REPRESENTATION, $galley->getId(), $submission->getId(), SUBMISSION_FILE_PROOF);
+			// filter supp files with DOI
+			$genreDao = DAORegistry::getDAO('GenreDAO');
+			$suppGalleyFiles = array();
+			foreach ($galleyFiles as $galleyFile) {
+				$genre = $genreDao->getById($galleyFile->getGenreId());
+				if ($genre->getSupplementary()) {
+					if ($galleyFile->getStoredPubid('doi')) {
+						$suppGalleyFiles[] = $galleyFile;
+					}
+				} else {
+					$submissionGalleyFiles[$galley->getBestGalleyId()] = $galleyFile;
+				}
+			}
+			if (!empty($suppGalleyFiles)) {
+				// construct the array key with galley best ID and locale needed for the component node
+				$componentFiles[$galley->getBestGalleyId().'-'.$galley->getLocale()] = $suppGalleyFiles;
+			}
+		}
+		// submission galley files - colelction nodes
+		if (!empty($submissionGalleyFiles)) {
+			$this->appendCollectionNodes($doc, $doiDataNode, $submission, $submissionGalleyFiles);
+		}
+		$journalArticleNode->appendChild($doiDataNode);
+
+		// component list (supplementary files)
+		if (!empty($componentFiles)) {
+			$journalArticleNode->appendChild($this->createComponentListNode($doc, $submission, $componentFiles));
+		}
+
+		return $journalArticleNode;
+	}
+
+	/**
+	 * Append all collection nodes 'collection' to the doi data node.
+	 * @param $doc DOMDocument
+	 * @param $doiDataNode DOMElement
+	 * @param $submission PublishedArticle
+	 * @param $submissionFiles array (best galley ID => submission file)
+	 */
+	function appendCollectionNodes($doc, $doiDataNode, $submission, $submissionFiles) {
+		$deployment = $this->getDeployment();
+		$context = $deployment->getContext();
+		$request = Application::getRequest();
+
+		// start of the text-mining collection element
+		$textMiningCollectionNode = $doc->createElementNS($deployment->getNamespace(), 'collection');
+		$textMiningCollectionNode->setAttribute('property', 'text-mining');
+		foreach ($submissionFiles as $betGalleyId => $submissionFile) {
+			$resourceURL = $request->url($context->getPath(), 'article', 'download', array($submission->getBestArticleId(), $betGalleyId, $submissionFile->getFileId()));
```

Comment:
This is I believe for DB queries, but when you call it on the object itself, it gets its data...